### PR TITLE
Finished quiz

### PR DESCRIPTION
**Pull request by Cajam**
Quiz complete

### CODE REVIEW

**Comment by Cajam** [comment]
yup

File: README.md [hunk 1]
```diff
@@ -25,8 +25,11 @@ an old beggar, X?", where X is the person passed into the method.
 Demonstrate calling the method with an argument of "young prince".
 
 Write your code here:
-```ruby
-# code here
+```
+def offerRose(X)
```
Comment (jshawl) [inline README.md:29]: avoid uppercase variable names when possible, but this is right!

File: README.md [hunk 2]
```diff
@@ -25,8 +25,11 @@ an old beggar, X?", where X is the person passed into the method.
...
+  "Would you take this rose and help out an old beggar, " + X + "?"
```
Comment (jshawl) [inline README.md:30]: can also use interpolation: `"...beggar, #{X}?"`

File: README.md [hunk 3]
```diff
@@ -48,8 +51,9 @@ Using Ruby, remove Belle from the town residents, and
 add her to the list of guests in the castle.
 
 Write your code here:
-```ruby
-# code here
+```
+residents.delete("Belle")
```
Comment (jshawl) [inline README.md:55]: `town[:residents]`, and try saving this to a variable. http://ruby-doc.org/core-2.2.0/Array.html#method-i-delete

File: README.md [hunk 4]
```diff
@@ -48,8 +51,9 @@ Using Ruby, remove Belle from the town residents, and
...
+castle[:residents] = "Belle"
```
Comment (jshawl) [inline README.md:56]: `town[:castle][:residents]`

File: README.md [hunk 5]
```diff
@@ -70,8 +74,10 @@ Belle is friends with Mrs. Potts
 ```
 
 Write your code here:
-```ruby
-# code here
+```
+friends.each do |name|
+puts "Belle is friends with #{name}"
+end
```
Comment (jshawl) [inline README.md:80]: :+1:

File: README.md [hunk 6]
```diff
@@ -89,7 +95,7 @@ entities (no need to draw an ERD):
 
 Your answer:
 ```
-Replace this with your answer
+An ERD is a model that demonstrates how different objects and classes and features relate to each other in a program. There is one genie to one lamp. One lamp can have many people that rub it. One person may have many pets. One pet has one person as an owner.
```
Comment (jshawl) [inline README.md:98]: or.. a visual representation of data relationships

File: README.md [hunk 7]
```diff
@@ -100,7 +106,18 @@ SQL database. If you need an example, you can use: people and wishes
 
 Your answer:
 ```
-Replace this with your answer
+The schema is an Active Record file that defines the key value pairs in a database.
+
+create table people (
+  id SERIAL PRIMARY KEY,
+  name
```
Comment (jshawl) [inline README.md:113]: what is the data type of `name`?

File: README.md [hunk 8]
```diff
@@ -100,7 +106,18 @@ SQL database. If you need an example, you can use: people and wishes
...
+);
+
+create table wishes (
+  id SERIAL PRIMARY KEY,
+  people_id
```
Comment (jshawl) [inline README.md:118]: what is the data type of `people_id`?

File: README.md [hunk 9]
```diff
@@ -124,8 +141,13 @@ Write code to do the following:
 
 
 Write your code here:
-```ruby
-# code here
+```
+lamp1 = Lamp.new(3)
```
Comment (jshawl) [inline README.md:145]: or `Lamp.new(wishes_remaining: 3)`

File: README.md [hunk 10]
```diff
@@ -124,8 +141,13 @@ Write code to do the following:
...
+lamp1.wishes_remaining = 1
+jafar = Genie.new("Jafar")
+jafar.lamp1
```
Comment (jshawl) [inline README.md:148]: or `jarfar.lamp = lamp1`

File: README.md (hunk 8 above)
Comment (Cajam) [inline README.md:118]: A string?

Very respectfully,

_Cassidy J. Harpster_
cassidy.j.harpster@gmail.com
(850) 218-4673
Washington, D.C.

File: README.md [hunk 11]
```diff
@@ -149,7 +171,17 @@ The ancestors have provided an example of one route; you do the other six!
...
+Creates a new warrior
+
+PUT '/warriors/1'
+Updates all of a warrior's information
+
+PATCH '/warriors/1'
+Updates new information in a warrior instead of all of it like with PUT
+
+DELETE '/warriors/1'
+Deletes this warrior
```
Comment (jshawl) [inline README.md:184]: only missing `GET /warriors/1/edit` and `GET /warriors/new`

File: README.md [hunk 12]
```diff
@@ -140,7 +162,7 @@ would look like for such an application.
 
 Your description:
 ```
-Replace this with your answer
+RESTful routes are routes that use 5 different methods for a browser to make requests to a server.
```
Comment (jshawl) [inline README.md:165]: Try using the word "convention" in your answer

File: README.md [hunk 13]
```diff
@@ -169,6 +201,16 @@ end
...
+
+<ul>
+  <% @warriors.each do |warrior| %>
+    <li>
+      <a href="/warriors/<%= warrior.id %>">
+        <%= warrior.title %>
+      </a>
+    </li>
+  <% end %>
+</ul>
```
Comment (jshawl) [inline README.md:215]: :+1:

**Comment by jshawl** [comment]
:+1: There are a few things in this quiz I recommend you go back and review - specifically accessing things in a hash and associating active record objects. Please let me know if you'd like to work through anything together

**Pull request closed without merging**